### DISCUSSION TITLE
Link with query parameter option

### DISCUSSION THREAD

**Discussion by yingban** [craftcms/ckeditor#209] [Ideas] [answered]
Hi,
I would like to know if it would be possible to add an option in the link feature to enable users for adding a query parameter or a hash to an URL in CKEditor fields?

**What is the expected behavior of the proposed feature?**
On link creation, if the option is set to true, there would be a second text field which must start with "#" or "?".
It will be automatically added the URL.

I already created a feature request on CKEditor's github, but it does not seem to gather enough interest.

**Comment by brandonkelly** [comment, marked as answer]
If the URL was created via “Link to an entry”, you can add a custom URL fragment _before_ the `#entry:…` one that is added automatically. For example:

```
https://mysite.com/linked/entry/uri#custom-fragment#entry:4@1:url
```

On the front-end, that will result in:

```
https://mysite.com/linked/entry/uri#custom-fragment
```

If you need to remove the `#entry:…` fragment altogether. At which point the URL will just be saved to the field data verbatim, as if you had entered the URL manually via “Insert link”.

```
https://mysite.com/linked/entry/uri?foo=bar
```

(Note that CKEditor 4.0.0 - 4.0.3 had a bug where custom URL fragments weren’t outputting correctly on the front-end, so if this is for Craft 5/CKEditor 4, you will need to update to CKEditor 4.0.4 for that to work.)

**Comment by yingban** [comment]
Thanks @brandonkelly!
